app/models/item.rb

```diff
@@ -1,3 +1,10 @@
 class Item < ApplicationRecord
+  validates_presence_of :name, :price
+
   belongs_to :supermarket
+  has_and_belongs_to_many :customers, join_table: 'customer_items'
+
+  def customer_count
+    self.customers.count
```

Comment:
Excellent work encapsulating this in a model method! Even though this is a simple data calculation, it's still a data calculation so to adhere to MVC strictly, this is the best place for it to be!